diffrax/_integrate.py

```diff
@@ -54,7 +59,7 @@ class SaveState(eqx.Module):
     save_index: IntScalarLike
 
 
-class State(eqx.Module):
+class State(eqx.Module, Generic[ProgressBarGenericState]):
```

Comment:
I'd probably remove the `Generic` here, and just annotate `progress_bar_state: PyTree[Array]`. This will
(a) avoid the `ProgressBarGenericState` complexity -- imagine how highly parameterised this class would be if we included all of them! -- and
(b) means that at test-time, when we do runtime type checking, we'll more carefully check that this attribute consists only of arrays.

(Also, note how I'm suggesting calling it `progress_bar_state` -- I think verbosity is better here, for something that isn't used very much.)